code/js/rcalc-keypress.js

```diff
@@ -0,0 +1,107 @@
+var isSecondPressed = false;
+var collectionOfKeys = [];
+var nextToLastCharacter;
+
+document.addEventListener("keyup", function(event) {
+  if (event.defaultPrevented) {
+    return;
+  }
+  var key = event.key;
+  collectionOfKeys.push(key);
+  nextToLastCharacter = collectionOfKeys[collectionOfKeys.length - 2];
+  switch (key) {
+    case "0":
```

Comment:
For many keys you are do same thing. You don't have to write same code, just write list of keys and then same action.
case "1":
case "2":
your action;
break;

You save many rows.

Reply:
keys are listed as adviced